sdks/python/apache_beam/transforms/external.py

```diff
@@ -42,6 +46,103 @@
   grpc = None
 # pylint: enable=wrong-import-order, wrong-import-position, ungrouped-imports
 
+DEFAULT_EXPANSION_SERVICE = 'localhost:8097'
+
+
+def iter_urns(coder, context=None):
+  yield coder.to_runner_api_parameter(context)[0]
+  for child in coder._get_component_coders():
+    for urn in iter_urns(child, context):
+      yield urn
+
+
+class PayloadBuilder(object):
+  """
+  Abstract base class for building payloads to pass to ExternalTransform.
+  """
+
+  @classmethod
+  def _config_value(cls, obj, typehint=None):
+    """
+    Helper to create a ConfigValue with an encoded value.
+    """
+    if typehint is None:
+      coder = registry.get_coder(type(obj))
+    else:
+      coder = registry.get_coder(typehint)
+    return ConfigValue(
+      coder_urn=list(iter_urns(coder)),
+      payload=coder.encode(obj))
+
+  def build(self):
+    """
+    :return: ExternalConfigurationPayload
+    """
+    raise NotImplementedError
+
+
+class SchemaBasedPayloadBuilder(PayloadBuilder):
+  """
+  Base class for building payloads based on a schema that provides
+  type information for each configuration value to encode.
+  """
+
+  def __init__(self, values, schema=None):
+    self._values = values
+    self._schema = schema or {}
+
+  @classmethod
+  def _encode_config(cls, config, schema):
+    result = {}
+    for k, v in config.items():
+      typehint = schema.get(k)
+      if v is None and (
+          typehint is None or not isinstance(typehint, typehints.Optional)):
```

Comment:
So seems like 'None' here means, used default value in remote SDK, right ? We should clarify that in documentation.

Reply:
The behavior is a bit undefined at this point, since we only have `GenerateSequence` as an example.  In that case, the field is null on the Java side as well.  

The reason we skip the fields if they're none here is ultimately because we don't have a cross-platform coder for none/null or a cross-platform way of declaring a type as optional/nullable.  So the behavior would be better defined if we _could_ serialize None and send that value to Java.